From 86ca83291d964deb47723667ba26c1497b3a4b4d Mon Sep 17 00:00:00 2001
From: Daniel Woodworth <daniel.woodworth@intel.com>
Date: Mon, 13 Jun 2022 16:00:44 -0400
Subject: [PATCH 047/172] [NFC][Alignment] Use getAlign in SPIRVWriter

LLVM is removing the getAlignment method from LoadInst and StoreInst in
these commits:

* https://github.com/llvm/llvm-project/commit/8865700f905173b71932cc241b5d36600343f425
* https://github.com/llvm/llvm-project/commit/2b89a4dc51d539cefd3628775883a32c9dd6fc64

This change fixes build errors with these LLVM changes by replacing the
remaining uses of getAlignment in SPIRVWriter with getAlign.

(cherry picked from commit e4862b74c8157b0192385a760392a39afabdd48c)
---
 lib/SPIRV/SPIRVWriter.cpp | 12 ++++--------
 1 file changed, 4 insertions(+), 8 deletions(-)

diff --git a/lib/SPIRV/SPIRVWriter.cpp b/lib/SPIRV/SPIRVWriter.cpp
index b290f426..807df1a0 100644
--- a/lib/SPIRV/SPIRVWriter.cpp
+++ b/lib/SPIRV/SPIRVWriter.cpp
@@ -1639,10 +1639,8 @@ LLVMToSPIRVBase::transValueWithoutDecoration(Value *V, SPIRVBasicBlock *BB,
     std::vector<SPIRVWord> MemoryAccess(1, 0);
     if (ST->isVolatile())
       MemoryAccess[0] |= MemoryAccessVolatileMask;
-    if (ST->getAlignment()) {
-      MemoryAccess[0] |= MemoryAccessAlignedMask;
-      MemoryAccess.push_back(ST->getAlignment());
-    }
+    MemoryAccess[0] |= MemoryAccessAlignedMask;
+    MemoryAccess.push_back(ST->getAlign().value());
     if (ST->getMetadata(LLVMContext::MD_nontemporal))
       MemoryAccess[0] |= MemoryAccessNontemporalMask;
     if (MDNode *AliasingListMD = ST->getMetadata(LLVMContext::MD_alias_scope))
@@ -1670,10 +1668,8 @@ LLVMToSPIRVBase::transValueWithoutDecoration(Value *V, SPIRVBasicBlock *BB,
     std::vector<uint32_t> MemoryAccess(1, 0);
     if (LD->isVolatile())
       MemoryAccess[0] |= MemoryAccessVolatileMask;
-    if (LD->getAlignment()) {
-      MemoryAccess[0] |= MemoryAccessAlignedMask;
-      MemoryAccess.push_back(LD->getAlignment());
-    }
+    MemoryAccess[0] |= MemoryAccessAlignedMask;
+    MemoryAccess.push_back(LD->getAlign().value());
     if (LD->getMetadata(LLVMContext::MD_nontemporal))
       MemoryAccess[0] |= MemoryAccessNontemporalMask;
     if (MDNode *AliasingListMD = LD->getMetadata(LLVMContext::MD_alias_scope))
-- 
2.20.1

